FirestoreWrite.ts

```diff
@@ -30,16 +30,19 @@ class FirestoreWrite {
    * @param {string} path the path of the document to update
    * @param {object} fields the document's new fields
    * @param {string} request the Firestore Request object to manipulate
-   * @param {boolean} mask if true, the update will use a mask. i.e. true: updates only specific fields, false: overwrites document with specified fields
+   * @param {boolean|string[]} mask the update will mask the given fields,
+   * if is an array (of field names), that array would be used as the mask. i.e. true: updates only specific fields, false: overwrites document with specified fields
+   * see jsdoc of the `updateDocument` method in Firestore.ts for more details
    * @return {object} the Document object written to Firestore
    */
   updateDocument_(path: string, fields: Record<string, any>, request: Request, mask = false): Document {
     if (mask) {
+      const maskData = typeof mask === 'boolean' ? Object.keys(fields) : mask;
```

Comment:
It would be better to check if mask is an array, because if the user passes an object, or an int, rest of the process would fail miserably.

Reply:
Any issues with this?

Reply:
I don't see it. I was referring specifically to this line to change the check of `typeof mask === 'boolean'` specifically.

Reply:
Take a look at the [next line](https://github.com/grahamearley/FirestoreGoogleAppsScript/blob/909c771c580af547b9a5817725065939357076e3/FirestoreWrite.ts#L43) 😉

Reply:
I was thinking more along the lines of defaulting to no mask (all keys in `fields`) if it wasn't the right type, thoughts?

Reply:
Hmmm, I don't think it's the best way to handle it. If the user enters some random value, you don't want it to default to some behavior that the user might not be aware of. I think it would be better to throw an error and let the user fix his mask parameter.

Reply:
What do you think?